app/controllers/application_controller.rb

```diff
@@ -755,10 +755,9 @@ def schedule_enable_disable(schedules, enabled)
   # Build the user_emails hash for edit screens needing the edit_email view
   def build_user_emails_for_edit
     @edit[:user_emails] = {}
-    User.all.sort_by { |u| u.name.downcase }.each do |u|
-      unless u.email.blank? ||
-             (@edit[:new][:email][:to] && @edit[:new][:email][:to].include?(u.email))
-        @edit[:user_emails][u.email] = "#{u.name} (#{u.email})"
+    User.with_current_user_groups.uniq.sort_by { |u| u.name.downcase }.each do |u|
+      unless @edit[:new][:email][:to] && @edit[:new][:email][:to].include?(u.email)
+        @edit[:user_emails][u.email] = "#{u.name} (#{u.email})" unless u.email.blank?
```

Comment:
yes, definitely better, I just removed line ```next unless to_email``` and I initialized ```to_email``` to ```[]``` when it is nil (```to_email = @edit[:new][:email][:to] || []```)